src/Negotiation/AbstractNegotiator.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Negotiation;
+
+abstract class AbstractNegotiator
+{
+    /**
+     * @param string $header     A string containing an `Accept|Accept-*` header.
+     * @param array  $priorities A set of server priorities.
+     *
+     * @return BaseAccept best matching type
+     */
+    public function getBest($header, array $priorities)
+    {
+        if (!$priorities) {
+            throw new \Exception('no priorities given'); 
```

Comment:
`\InvalidArgumentException` seems appropriate.

`\ParseHeaderException` and `\ParseTypeException` can inherit from `\ParseException`.